core-it-suite/src/test/java/org/apache/maven/it/MavenITmng2339BadProjectInterpolationPre4Test.java

```diff
@@ -0,0 +1,75 @@
+package org.apache.maven.it;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.maven.it.util.ResourceExtractor;
+
+import java.io.File;
+
+/**
+ * This is a test set for <a href="https://issues.apache.org/jira/browse/MNG-2339">MNG-2339</a>.
+ *
+ *
+ */
+public class MavenITmng2339BadProjectInterpolationPre4Test
```

Comment:
This class isn't registered with the suite, how should this run?

Reply:
It would not. But adding it to the suite would, so I've pushed an additional commit just for that. Thanks for pointing that out. Double checked: running the IT's with Maven 3.8.5 runs both test classes for MNG-2339, running with Maven 4.0.0-alpha-1-SNAPSHOT runs only one and skips this newly added one.

Reply:
Have to considered to keep it in the original class and use this approach: https://github.com/apache/maven-integration-testing/blob/7fb56604e6e74afe293a42e82f20665566f10ba8/core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4768NearestMatchConflictResolutionTest.java#L52?

Reply:
Oh, cool, I didn't know that one! So no, certainly didn't consider that, but I'd say it's cleaner indeed (doesn't introduce a new class, etc.). I've restored the original situation with just that `requiresMavenVersion(...)` added. Thanks for the suggestion!